src/sentry/rules/processor.py

```diff
@@ -84,7 +85,7 @@ def get_rules(self) -> Sequence[Rule]:
         return rules_
 
     def _build_rule_status_cache_key(self, rule_id: int) -> str:
-        return "grouprulestatus:1:%s" % hash_values([self.group.id, rule_id])
+        return "grouprulestatus:1:%s" % hash_values([self.group.id, rule_id])  # type: ignore[union-attr]
```

Comment:
this indicates the change above was not correct -- probably the other caller needs to convert to GroupEvent from Event (ensuring that `group` is always present)

again `# type: ignore` should only be used for bugs in mypy -- it should not be used to "fix" the type checker like this.  the type checker is telling you that this is unsafe as written (because it is! -- an Event does not always have a group -- but a GroupEvent does!)

Reply:
thanks for the feedback! i'm gonna revert processor.py so that `event` is always a GroupEvent and instead address the typing errors in the calls to RuleProcessor